### DISCUSSION TITLE
Tremolo repetition dots & tuplet tremolos

### DISCUSSION THREAD

**Discussion by craigsapp** [music-encoding/music-encoding#954] [Q&A] [answered]
How should repetition dots on tremolos be encoded?  For example:

<img width="879" alt="Screen Shot 2022-04-30 at 3 02 55 PM" src="https://user-images.githubusercontent.com/3487289/166124093-f81b69bc-8f1b-4c91-b5fb-a63e438140f3.png">

The dots are not staccatos, but rather are used to indicate how many times the tremolo note/chord should be repeated (and typically just give on the first tremolo group to clarify the number of repetitions for the rest of the tremolos after it as in this case).

Test file:

<img width="389" alt="Screen Shot 2022-04-30 at 2 48 47 PM" src="https://user-images.githubusercontent.com/3487289/166123763-a3be671a-0754-409a-ab9c-651a7a5a5f7b.png">

<details>
<summary>
Click to view MEI data
</summary>

```xml
<?xml version="1.0" encoding="UTF-8"?>
<?xml-model href="https://music-encoding.org/schema/dev/mei-all.rng" type="application/xml" schematypens="http://relaxng.org/ns/structure/1.0"?>
<?xml-model href="https://music-encoding.org/schema/dev/mei-all.rng" type="application/xml" schematypens="http://purl.oclc.org/dsdl/schematron"?>
<mei xmlns="http://www.music-encoding.org/ns/mei" meiversion="5.0.0-dev">
 <meiHead>
  <fileDesc>
   <titleStmt>
    <title />
   </titleStmt>
   <pubStmt />
  </fileDesc>
  <encodingDesc>
   <appInfo>
    <application isodate="2022-04-30T14:49:25" version="3.10.0-dev-36a3430-dirty">
     <name>Verovio</name>
     <p>Transcoded from Humdrum</p>
    </application>
   </appInfo>
  </encodingDesc>
  <workList>
   <work>
    <title />
   </work>
  </workList>
 </meiHead>
 <music>
  <body>
   <mdiv xml:id="mot75oq">
    <score xml:id="sors0fo">
     <scoreDef xml:id="ssvpuwy">
      <staffGrp xml:id="siojni6">
       <staffDef xml:id="syav30v" n="1" lines="5">
        <clef xml:id="clef-L2F1" shape="F" line="4" />
       </staffDef>
      </staffGrp>
     </scoreDef>
     <section xml:id="section-L1F1">
      <measure xml:id="measure-L1" n="1">
       <staff xml:id="s7osgzg" n="1">
        <layer xml:id="layer-L1F1N1" n="1">
         <note xml:id="note-L5F1" dur="4" oct="3" pname="g" accid.ges="n" />
         <bTrem xml:id="btrem-L6F1-L9F1" unitdur="16">
          <note xml:id="note-L6F1" dur="4" oct="3" pname="d" accid.ges="n" />
         </bTrem>
         <bTrem xml:id="btrem-L10F1-L13F1" unitdur="16">
          <note xml:id="note-L10F1" dur="4" oct="3" pname="f" accid.ges="n" />
         </bTrem>
        </layer>
       </staff>
      </measure>
     </section>
    </score>
   </mdiv>
  </body>
 </music>
</mei>
```

</details>

Ideally this would be some sort of attribute on bTrem:
https://music-encoding.org/guidelines/v4/elements/btrem.html
but I do not see any current attributes which might be used for this (perhaps call a new attribute `bTrem@repdots` since `bTrem@dots` is used for duration of tremolo.

Note that these are not note articulations and should not be encoded as `note@artic`.  And in any case, I do not see any note articulations that match to this symbol.  

In this case there are four dots, but three dots is also common.  I may have rarely seen two and six as well.

Related to this, a number can be used, as in this example:

<img width="828" alt="Screen Shot 2022-04-30 at 3 00 39 PM" src="https://user-images.githubusercontent.com/3487289/166124026-933df826-540b-4c9a-844e-cdfddac900b3.png">

https://s9.imslp.org/files/imglnks/usimg/7/7f/IMSLP109732-PMLP25878-Schubert_Erlkonig_auto.pdf

Perhaps `bTrem@num` can be used for this?   But these are also the same as a tuplet number (6 in the time of 4).

When I try to use tuplets for the tremolo, I get the tuplet number, but the tremolo beam is not being drawn in verovio.  Is the following encoding correct?:

<img width="461" alt="Screen Shot 2022-04-30 at 3 13 43 PM" src="https://user-images.githubusercontent.com/3487289/166124360-51f77781-99e7-4e52-b740-980b8a17df78.png">

<details>
<summary>
Click to view MEI data for above example
</summary>

```xml
<?xml version="1.0" encoding="UTF-8"?>
<?xml-model href="https://music-encoding.org/schema/dev/mei-all.rng" type="application/xml" schematypens="http://relaxng.org/ns/structure/1.0"?>
<?xml-model href="https://music-encoding.org/schema/dev/mei-all.rng" type="application/xml" schematypens="http://purl.oclc.org/dsdl/schematron"?>
<mei xmlns="http://www.music-encoding.org/ns/mei" meiversion="5.0.0-dev">
 <meiHead>
  <fileDesc>
   <titleStmt>
    <title />
   </titleStmt>
   <pubStmt />
  </fileDesc>
  <encodingDesc>
   <appInfo>
    <application isodate="2022-04-30T15:09:10" version="3.10.0-dev-36a3430-dirty">
     <name>Verovio</name>
     <p>Transcoded from Humdrum</p>
    </application>
   </appInfo>
  </encodingDesc>
  <workList>
   <work>
    <title />
   </work>
  </workList>
 </meiHead>
 <music>
  <body>
   <mdiv xml:id="m9d9fdt">
    <score xml:id="sab6nck">
     <scoreDef xml:id="so60z8t">
      <staffGrp xml:id="s5bhhvq">
       <staffDef xml:id="spagukq" n="1" lines="5">
        <clef xml:id="cbiq38e" shape="G" line="2" />
        <meterSig xml:id="metersig-L2F1" count="4" unit="4" />
       </staffDef>
      </staffGrp>
     </scoreDef>
     <section xml:id="section-L1F1">
      <measure xml:id="measure-L1" n="1">
       <staff xml:id="s9j94o0" n="1">
        <layer xml:id="layer-L1F1N1" n="1">
         <tuplet xml:id="tuplet-L5F1-L10F1" num="6" numbase="4" bracket.place="below" bracket.visible="false" num.format="count">
          <beam xml:id="beam-L5F1-L10F1">
           <note xml:id="note-L5F1" dur="8" oct="4" pname="e" accid.ges="n" />
           <note xml:id="note-L6F1" dur="8" oct="4" pname="e" accid.ges="n" />
           <note xml:id="note-L7F1" dur="8" oct="4" pname="e" accid.ges="n" />
           <note xml:id="note-L8F1" dur="8" oct="4" pname="e" accid.ges="n" />
           <note xml:id="note-L9F1" dur="8" oct="4" pname="e" accid.ges="n" />
           <note xml:id="note-L10F1" dur="8" oct="4" pname="e" accid.ges="n" />
          </beam>
         </tuplet>
         <tuplet xml:id="tuplet-L5F1-L10F1" num="6" numbase="4" num.format="count">
            <bTrem xml:id="btrem-L12F1-L17F1" unitdur="4">
               <note xml:id="note-L12F1" dur="2" oct="4" pname="e" accid.ges="n" />
            </bTrem>
         </tuplet>
        </layer>
       </staff>
      </measure>
     </section>
    </score>
   </mdiv>
  </body>
 </music>
</mei>
```

</details>

**Comment by rettinghaus** [comment, marked as answer]
What did you expect by setting `@unitdur="4"`? With the correct value it works nicely 😜 
![trem](https://user-images.githubusercontent.com/7693447/166203525-45611f77-f66f-4db1-9f93-2f2ef7e1d7cd.png)

Often it is a good idea to have a look in the [Guidelines](https://music-encoding.org/guidelines/v4/content/cmn.html#cmnTrem) where `bTrem@num` is explained. (However, this is not implemented in Verovio yet.)

For specifying dots the `@dots` attribute would come to mind, but that would bring a quite substantial remodeling. As these are dots of division maybe we could add a `@divdots` attribute to `bTrem`. The follow-up question would be if it should allow all positive integer values (like `@num`) or if it should be restricted to "2", "3", "4" and "6"?

**Reply by craigsapp** [reply]
> With the correct value it works nicely 😜

I get it, but you should answer more explicitly for other people looking at this with a similar question. 8️⃣ 😛

**Reply by lpugin** [reply]
> As these are dots of division maybe we could add a `@divdots` attribute to bTrem. The follow-up question would be if it should allow all positive integer values (like `@num`) or if it should be restricted to "2", "3", "4" and "6"?

I would not call them dots of division. At least I think that would be confusing - thinking about mensural division dots. Maybe simply `@btrem.dots`? Also, I do not see a good reason for restricting the values to "2", "3", "4" and "6".

**Reply by kepper** [reply]
Even though I can't remember an example with `fTrem`s having such dots, I'd be inclined to call them `@trem.dots`, but maybe something different is even more comprehensible – what about `@visual.dots`? I agree on open (positive integer-based) values…

**Comment by musicEnfanthen** [comment]
Closing
